src/compile/data/stack.ts

```diff
@@ -152,15 +207,14 @@ export class StackNode extends DataFlowNode {
       }
       return [vgField(dimensionFieldDef)];
     }
-    return [];
+    return groupby || [];
```

Comment:
I'm ok with this for now.  

Once you decouple Impute from stacking (), we should make `dimensionFieldDef` no longer a part of StackComponent and just have `StackNode.makeFromEncoding()` create a `groupby`, using this method.  At that point, we can basically remove `facetBy` and just have push new dimension in the facet to `groupby` instead.

(Basically, if you look at the code, the final groupby in the stack transform = groupBy that's from dimension fieldDef + facetBy)